src/test/java/org/jenkins/plugins/lockableresources/LockStepTest.java

```diff
@@ -516,6 +519,48 @@ public void evaluate() throws Throwable {
 		});
 	}
 
+	@Issue("JENKINS-40368")
+	@Test
+	public void hardKillWithWaitingRuns() throws Exception {
+		story.addStep(new Statement() {
+			@Override
+			public void evaluate() throws Throwable {
+				LockableResourcesManager.get().createResource("resource1");
+				WorkflowJob p = story.j.jenkins.createProject(WorkflowJob.class, "p");
+				p.setDefinition(new CpsFlowDefinition("retry(99) {\n" +
+						"    lock('resource1') {\n" +
+						"        semaphore('wait-inside')\n" +
+						"     }\n" +
+						"}", true));
+
+				WorkflowRun prevBuild = null;
+				for (int i = 0; i < 3; i++) {
+					WorkflowRun rNext = p.scheduleBuild2(0).waitForStart();
+					if (prevBuild != null) {
+						story.j.waitForMessage("[resource1] is locked, waiting...", rNext);
+						Executor ex = prevBuild.getExecutor();
+						assertNotNull(ex);
+						ex.interrupt();
+						story.j.waitForMessage("Click here to forcibly terminate running steps", prevBuild);
+						prevBuild.doTerm();
+						story.j.waitForMessage("Click here to forcibly kill entire build", prevBuild);
```

Comment:
No, I mean that #46 makes this plugin to work properly when a situation caused by some other bug (maybe jenkinsci/workflow-basic-steps-plugin#35) arises.